nest/core/decorators/controller.py

```diff
@@ -1,47 +1,45 @@
 from typing import Optional, Type
 
-from fastapi.routing import APIRouter
-
-from nest.core.decorators.class_based_view import class_based_view as ClassBasedView
 from nest.core.decorators.http_method import HTTPMethod
 from nest.core.decorators.utils import get_instance_variables, parse_dependencies
 
 
 def Controller(prefix: Optional[str] = None, tag: Optional[str] = None):
     """
-    Decorator that turns a class into a controller, allowing you to define
-    routes using FastAPI decorators.
+    Decorator that marks a class as a controller, collecting route metadata
+    for future registration in the underlying framework.
 
     Args:
         prefix (str, optional): The prefix to use for all routes.
-        tag (str, optional): The tag to use for OpenAPI documentation.
+        tag (str, optional): The tag to use for grouping or doc generation.
 
     Returns:
-        class: The decorated class.
+        class: The decorated class (with route metadata added).
     """
 
-    # Default route_prefix to tag_name if route_prefix is not provided
     route_prefix = process_prefix(prefix, tag)
 
-    def wrapper(cls: Type) -> Type[ClassBasedView]:
-        router = APIRouter(tags=[tag] if tag else None)
-
-        # Process class dependencies
+    def wrapper(cls: Type) -> Type:
+        # 1. Process class-level dependencies
         process_dependencies(cls)
 
-        # Set instance variables
+        # 2. Set instance variables for any non-injected fields
         set_instance_variables(cls)
 
-        # Ensure the class has an __init__ method
+        # 3. Ensure the class has an __init__ method
         ensure_init_method(cls)
 
-        # Add routes to the router
-        add_routes(cls, router, route_prefix)
+        # 4. Gather route metadata
+        route_definitions = collect_route_definitions(cls, route_prefix)
+
+        # 5. Store routes in class attribute for later usage
+        setattr(cls, "__pynest_routes__", route_definitions)
```

Comment:
Magic string, should create reflection class (str_enum) for all __pynest_routes_ magic attributes (__injectable__ for example) and refrence it